app/jobs/reactions/reaction_create_job.rb

```diff
@@ -0,0 +1,14 @@
+module Reactions
+  class ReactionCreateJob < ApplicationJob
+    queue_as :reaction_create
+
+    def perform(user_id:, reactable_id:, reactable_type:, category:)
+      Reaction.create(
```

Comment:
I think this job should load both the user and the reactable from the DB and exit if any of them is not available.

You can check `app/jobs/articles/update_analytics_job.rb` ad an example.

This way the job will just exit if in the meantime those resources have disappeared.

Reply:
@rhymes there's something I don't understand about the reactable. From what I've looked up, I can get a reactable like this:

```ruby
reaction = Reaction.find_by(reactable_id: reactable_id, reactable_type: reactable_type)
reactable = reaction&.reactable
```

My job is essentially creating a new ``Reaction``. If I load the reactable this way, then to create a new ``Reaction`` using my job there must be another ``Reaction`` in the DB. That does not make sense to me. Am I missing something?

Reply:
You do have a point, I was thinking of loading the reactable without loading the reaction.

Something like `reactable_type.constantize.find(reactable is)` IIRC